src/backup.sh

```diff
@@ -0,0 +1,113 @@
+# This file deals with backing up and restoring previously backed up kw data
+# files stored in KW_DATA_DIR. These files are configs, pomodoro reports and
+# kw usage statistics.
+
+include "$KW_LIB_DIR/kwlib.sh"
+include "$KW_LIB_DIR/kwio.sh"
+
+declare -gA options_values
+
+# This is the main function that deals with the operations related to backup. It
+# recieves an argument and call other functions based on that.
+function backup()
+{
+  if [[ -z "$*" ]]; then
+    complain 'Please, provide an argument'
+    backup_help
+    exit 22 # EINVAL
+  fi
+
+  backup_parser "$@"
+
+  if [[ -n "${options_values['BACKUP_PATH']}" ]]; then
+    create_backup "${options_values['BACKUP_PATH']}"
+  fi
+
+  if [[ -n "${options_values['RESTORE_PATH']}" ]]; then
+    restore_backup "${options_values['RESTORE_PATH']}"
+  fi
+}
+
+# This function creates a .tar.gz file containing all the data found in
+# KW_DATA_DIR
+#
+# @path Path to the directory in which the compressed file should be stored.
+function create_backup()
+{
+  local path="$1"
+  local flag="$2"
+  local ret
+  local compress_cmd
+
+  if ! [[ -d "$path" ]]; then
+    complain "We couldn't find the path"
+    exit 2 # ENOENT
+  fi
+
+  compress_cmd="tar -C $KW_DATA_DIR -czf $path/backup.tar.gz ."
+  cmd_manager "$flag" "$compress_cmd"
+
+  ret="$?"
+  if [[ "$ret" != 0 ]]; then
+    complain "We couldn't create the tar file"
+    exit $ret
+  fi
```

Comment:
If the backup operation ends without any problem, let's tell the user by printing a successful message and tell them where the back can be found.